exp/examine/local.go

```diff
@@ -0,0 +1,164 @@
+package examine
+
+import (
+	"os"
+	"strings"
+
+	"github.com/chanzuckerberg/fogg/errs"
+	"github.com/hashicorp/go-getter"
+	"github.com/hashicorp/terraform-config-inspect/tfconfig"
+	"github.com/spf13/afero"
+)
+
+const apiHostname = "https://registry.terraform.io"
+const apiVersion = "/v1"
+const resourceType = "/modules/"
+
+//**Local refers to any files located within your local file system**
+
+//GetLocalModules retrieves all terraform modules within a given directory
+//TODO:(EC) Define local and global modules OR rename the values
+func GetLocalModules(fs afero.Fs, dir string) ([]ModuleWrapper, error) {
+	_, err := os.Stat(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	modules, err := getAllModules(fs, dir)
+	if err != nil {
+		return nil, err
+	}
+	return modules, nil
+}
+
+//getAllModules retrieves all modules (and their respective resources, data, etc...) being used by the directory
+func getAllModules(fs afero.Fs, dir string) ([]ModuleWrapper, error) {
+	var queue []ModuleWrapper
+
+	//Find all submodules for the current directory
+	submodules, err := getSubmodules(fs, dir)
+	if err != nil {
+		return nil, err
+	}
+	queue = append(queue, submodules...)
+
+	//BFS on the current queue of submodules to find their submodules
+	for { //Do: recursively add new submodules to queue, While: queue is not empty
+
+		submods, err := getSubmodules(fs, queue[0].module.Path)
+		if err == nil {
+			//Add new elements to temp
+			queue = append(queue, submods...)
+			submodules = append(submodules, submods...)
+		}
+		//remove the current element
+		queue = append(queue[:0], queue[1:]...)
+
+		if !(len(queue) > 0) {
+			break
+		}
+	}
+
+	return submodules, nil
+}
+
+//getSubmodules retrieves all submodules related to the root directory
+func getSubmodules(fs afero.Fs, dir string) ([]ModuleWrapper, error) {
```

Comment:
I think we should avoid trying to implement this ourselves, and link to the code in terraform itself to do it.